cs-quicklabs/quick-learn
# Feature archived

# Add Archive Functionality for Lessons and Courses

## Overview
This PR implements archive functionality for lessons and courses, including list views with infinite scroll pagination, restore capabilities, and proper user tracking.

## Changes

### Frontend Components
- Added `ArchivedLessons.tsx` component
  - Infinite scroll pagination
  - Search functionality
  - Restore/delete modal confirmations
  - User tracking for archive actions

- Updated API enum constants
```typescript
export enum ArchivedApiEnum {
  // Existing endpoints...
  ARCHIVED_LESSONS = '/lesson/archived',
  ACTIVATE_LESSON = '/lesson/activate',
  ARCHIVED_COURSES = '/course/archived',
  ACTIVATE_COURSE = '/course/activate',
}
```

- Added archive-related translations to `en.ts`
```typescript
archivedLessons: 'Archived Lessons',
archivedLessonsSubtext: 'Following lessons have been archived.',
confirmActivateLesson: 'Are you sure you want to activate this lesson?',
confirmActivateLessonSubtext: 'Once this lesson is restored, users will be able to see it and access its content.',
// ... similar for courses
```

### Backend Changes
- Added archive-related endpoints in LessonController
- Updated LessonEntity to track archive status and archive_by user
- Added pagination support for archived items
- Proper relation loading for course, user, and category data

### Bug Fixes
- Fixed pagination in archived views to properly handle total_pages
- Corrected hasMore logic for infinite scroll
- Fixed type mismatches between string and number IDs

## API Response Structure
```typescript
{
  success: boolean;
  message: string;
  data: {
    items: Array<TLesson | TCourse>;
    total: number;
    page: number;
    total_pages: number;
  }
}
```

## Testing Instructions
1. Archive a lesson/course
   - Verify it appears in archived list
   - Check user tracking info is correct

2. Test pagination
   - Scroll through list with >10 items
   - Verify all pages load correctly

3. Test restore functionality
   - Restore archived item
   - Verify it disappears from archived list
   - Check it's accessible in main view

4. Test search
   - Search in archived items
   - Verify pagination works with search

## Dependencies
- Frontend:
  - react-infinite-scroll-component
  - react-toastify (for notifications)

- Backend:
  - Updated TypeORM entities
  - NestJS pagination service

## Screenshots
![image](https://github.com/user-attachments/assets/0d2325ef-e8e8-44b1-9ad5-89ed8a4a37b1)
![image](https://github.com/user-attachments/assets/b066a437-8dc5-4739-9669-3639f9839b25)

## Notes
- The archive functionality maintains user tracking for both archive and restore actions
- Search is implemented across name, description, and category name
- Pagination is set to 10 items per page as per existing pattern
